### PR TITLE
Added a Setup and Contributing guide.

### PR DESCRIPTION
**Pull request by cngo-github**
Added guides for how to set-up an environment, how to use SBT in the command line, and how to contribute to the project.

TODO: Add instructions for setting up IntelliJ.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (marcinwadon) [inline CONTRIBUTING.md:3]: For community contributors, there is no way to label and request review for the pull request, so it must be done by someone with access to the main repository.

File: CONTRIBUTING.md
Comment (marcinwadon) [inline CONTRIBUTING.md:7]: It doesn't matter on which branch you develop, because you're working on the fork. Fork doesn't have protected branches as the main repository.

File: CONTRIBUTING.md
Comment (marcinwadon) [inline CONTRIBUTING.md:8]: I'd say that changes should be atomic (or as atomic as possible). That being said, sometimes it's fine to put multiple commits into a pull request, if they are meaningful (see: #218).

And by atomic changes, I mean something like this:
image

I could merge it with other changes in the same pull request, but then it would be impossible to revert it or lookup the history for that fix.

File: CONTRIBUTING.md
Comment (marcinwadon) [inline CONTRIBUTING.md:9]: I'd say that you should use `rebase`. Assuming that the remote is `origin` (you may name it differently for fork):
```
git fetch origin
git rebase origin/develop
git checkout my_branch
git rebase develop
```

Cherrypick is useful but not day-to-day, at least in my opinion.

File: CONTRIBUTING.md
Comment (marcinwadon) [inline CONTRIBUTING.md:11]: I wouldn't create draft PR by default. Only, if pull request needs additional changes and it's not considered finished but I want someone else to review it already. Otherwise, I would apply the pull request as ready for review.

File: CONTRIBUTING.md
Comment (marcinwadon) [inline CONTRIBUTING.md:12]: That's incorrect and we should remove that part. We cannot give an access to everyone in the community to manage pull requests and issues.

File: CONTRIBUTING.md
Comment (marcinwadon) [inline CONTRIBUTING.md:18]: typo

File: CONTRIBUTING.md
Comment (marcinwadon) [inline CONTRIBUTING.md:27]: `sbt test`

File: SETUP.md
Comment (marcinwadon) [inline SETUP.md:19]: Docker isn't required for contribution. It's an optional to spin up a local cluster using kubernetes. There are other ways as well.

File: SETUP.md
Comment (marcinwadon) [inline SETUP.md:20]: I'd link the docker documentation here, as the process may change and it's OS dependant. We shouldn't copy their documentation imo

File: CONTRIBUTING.md
Comment (kpudlik) [inline CONTRIBUTING.md:18]: `marcinwadon` (typo)

Comment (kpudlik) [reply CONTRIBUTING.md:18]: `review` (typo)

File: SETUP.md
Comment (marcinwadon) [inline SETUP.md:64]: That's wrong and may be insecure. Token should has least-required scopes - currently it's required to fetch the github package only (one of dependencies of the repository). Also, it's wise to set the expiration date and rotate tokens. I'd avoid writing down such things in the official repo.

**Review by kpudlik — CHANGES_REQUESTED** [review]

**Review by marcinwadon — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (marcinwadon) [reply CONTRIBUTING.md:12]: That's true. However, we will see the pull request anyway, so there is no need to mention anyone (if the pull request is made by community contributors)

File: CONTRIBUTING.md
Comment (ryle-goehausen) [reply CONTRIBUTING.md:27]: I'd add one more thing here, `sbt runLinter` which auto-formats the code according to style guide. This isn't documented.

File: SETUP.md
Comment (ryle-goehausen) [reply SETUP.md:19]: I would mark this as optional -- and make it clear it's for running local clusters and integration tests, I think it should be included as a default but made clear that it's not strictly required for IDE development, only cluster level testing.

File: SETUP.md
Comment (ryle-goehausen) [reply SETUP.md:64]: These steps are already documented externally: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token  Include a link to this instead of repeating.

Also the only scope that is required should be 
>    read:packages
>  Download packages from GitHub Package Registry

I believe, should confirm but that's what I have set.

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: Fixed: https://github.com/Constellation-Labs/tessellation/pull/266/commits/510f0859e09d3438f1569a368bf62276e1246cdd

Comment (cngo-github) [reply CONTRIBUTING.md]: Fixed: https://github.com/Constellation-Labs/tessellation/pull/266/commits/510f0859e09d3438f1569a368bf62276e1246cdd

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: Fixed typo: https://github.com/Constellation-Labs/tessellation/pull/266/commits/228868cde7b40cfd716dd767e3f1a7bb1374f048

Added `sbt runLinter`: https://github.com/Constellation-Labs/tessellation/pull/266/commits/0ac5619c761465906296ab416cc941b1e26bf39a

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: Updated the instructions to clarify this.

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: Removed intruction.

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: Changed to clarify that changes should be atomic.

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: Changed instructions to use `git rebase`.

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: Changed to create a regular PR.

File: SETUP.md
Comment (cngo-github) [reply SETUP.md]: Separated out Docker into its own section and clarified it as optional.

File: SETUP.md
Comment (cngo-github) [reply SETUP.md]: Replaced instructions with the provided link.

File: CONTRIBUTING.md
Comment (marcinwadon) [inline CONTRIBUTING.md:22]: If this will not be removed, please take a look at Ryle's comment related to that part

**Review by marcinwadon — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (ryle-goehausen) [reply CONTRIBUTING.md]: In general, we should rely on a more detailed contributing guide similar to https://github.com/apache/spark/blob/master/CONTRIBUTING.md -- but I don't think we need all of that yet since that's a much larger project.

**Review by ryle-goehausen — APPROVED** [review]
LGTM after changes, please await @marcinwadon approval as well assuming you have addressed remainder of his comments.

**Review by marcinwadon — COMMENTED** [review]
Looks fine for me as well, however point 8 and 9 in setup file (re: GitHub token) is required not only for the docker but for development in general (to fetch project dependencies). Can you move that part above docker, as one of the requirements to start working on the project?

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: The section has been removed.

File: SETUP.md
Comment (cngo-github) [reply SETUP.md]: Fixed.

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: The section has been removed.

File: CONTRIBUTING.md
Comment (cngo-github) [reply CONTRIBUTING.md]: Added instruction to mention a general category in the PR. The categories are:

- Bug Fix
- Feature
- Documentation
- New/updated tests

**Review by marcinwadon — APPROVED** [review]

**Review by kpudlik — APPROVED** [review]

**Pull request merged**
